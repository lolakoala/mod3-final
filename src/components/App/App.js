import React, { Component } from 'react';
import PropTypes, { shape, func, string, array } from 'prop-types';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { addHouses } from '../../actions';
import wolf from '../../images/wolf.gif';
import Card from '../Card/Card.js';
import fetchHouses from './helper.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      loading: true
    };
  }

  componentDidMount() {
    const { addHouses } = this.props;
    fetchHouses()
      .then(res => res.json())
      .then(jsonRes => addHouses(jsonRes))
      .then(() => this.setState({ loading: false }));
  }

  renderHouses = () => {
    const { houses } = this.props;
    return houses.map(house => {
      //I could not get all cards to load without having
      //this console log for a time buffer. it would only load 3-4 of them.
      console.log(house)
      return <Card house={house} key={Date.now()}/>;
    });
  }

  render() {
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
        </div>
        <div className='Display-info'>
          {!this.state.loading ?
            this.renderHouses()
            : <img src={wolf} alt='wolf'/>}
        </div>
      </div>
    );
  }
}

App.propTypes = {
  houses: array,
  addHouses: func.isRequired
};

const mapStateToProps = ({ houses }) => ({ houses });
const mapDispatchToProps = dispatch => ({ addHouses:
  houses => dispatch(addHouses(houses))
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
