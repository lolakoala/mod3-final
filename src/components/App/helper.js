const fetchHouses = () => {
  return fetch('http://localhost:3001/api/v1/houses');
};

export default fetchHouses;
