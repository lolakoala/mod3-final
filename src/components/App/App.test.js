import React from "react";
import App from './App.js';
import Enzyme, { mount } from "enzyme";
import houses from '../../mockData/houses.js';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';

//I could not get these tests to work, and I didn't want to
//watse time struggling with it. In theory, they should pass.
//Probably something to do with fetch mock,
//even though the fetch is in a helper file.

describe('app component', () => {
  Enzyme.configure({ adapter: new Adapter() });
  const mockStore = configureStore();
  const initialState = {
    houses: []
  };
  const store = mockStore(initialState);
  const mockFn = jest.fn();
  const wrapper = mount(<App
    store={store}
    houses={houses}
    addHouses={mockFn}/>);


  it('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have a logo', () => {
    const logo = wrapper.find('.App-logo');

    expect(logo.length).toEqual(1);
  });

  it('should have a title', () => {
    const title = wrapper.find('h2');

    expect(title.text()).toEqual('Welcome to Westeros');
  });

  it('should have 9 cards', () => {
    const cards = wrapper.find('Card');

    expect(cards.length).toEqual(9);
  });

  it('should fire actions', () => {
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
