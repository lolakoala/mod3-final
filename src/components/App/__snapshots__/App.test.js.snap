// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component should match snapshot 1`] = `undefined`;
